File: WpfGrafik/Zeichnung.vb
Imports wpf = System.Windows
Public Class Zeichnung
    Public Elemente As New List(Of wpf.Shapes.Shape)

    Public Sub zeichne(ByVal zeichenbrett As wpf.Controls.Canvas)
        zeichenbrett.Children.Clear()

        For Each sh In Elemente
            zeichenbrett.Children.Add(sh)
        Next
    End Sub

    Public Sub auswählen(ByVal ix As Integer)
        For i As Integer = 0 To Elemente.Count - 1
            If i = ix Then
                Elemente(i).Fill = Brushes.Red
                Elemente(i).Stroke = Brushes.Blue
            Else
                Elemente(i).Fill = Brushes.Transparent
                Elemente(i).Stroke = Brushes.Black
            End If
        Next
    End Sub

    Public Sub auswählen(ByVal pos As Point)
        For i As Integer = 0 To Elemente.Count - 1
            If Elemente(i).IsMouseOver Then
                'Elemente(i).Fill = Brushes.Red
                Elemente(i).Stroke = Brushes.Blue
            Else
                'Elemente(i).Fill = Brushes.Transparent
                Elemente(i).Stroke = Brushes.Black
            End If
        Next
    End Sub
End Class

File: Wpf2DGrafik/Zeichnung.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using wpf = System.Windows;
using media = System.Windows.Media;

namespace Wpf2DGrafik
{
    public class Zeichnung
    {
        public List<wpf.Shapes.Shape> Elemente = new List<wpf.Shapes.Shape>();

        public void zeichne(wpf.Controls.Canvas zeichenbrett)
        {
            zeichenbrett.Children.Clear();

            foreach (var sh in Elemente)
                zeichenbrett.Children.Add(sh);
        }


        public void auswählen(int ix)
        {
            for (int i = 0; i < Elemente.Count; i++)
            {
                if (i == ix)
                {
                    Elemente[i].Fill = media.Brushes.Red;
                    Elemente[i].Stroke = media.Brushes.Blue;
                }
                else
                {
                    Elemente[i].Fill = media.Brushes.Transparent;
                    Elemente[i].Stroke = media.Brushes.Black;
                }
            }
        }

        public void auswählen(wpf.Point pos)
        {
            for (int i = 0; i < Elemente.Count; i++)
            {
                if (Elemente[i].IsMouseOver)
                {
                    //Elemente(i).Fill = Brushes.Red
                    Elemente[i].Stroke = media.Brushes.Blue;
                }
                else
                {
                    //Elemente(i).Fill = Brushes.Transparent
                    Elemente[i].Stroke = media.Brushes.Black;
                }
            }
        }
    }
}
